addons/account_asset/account_asset.py

```diff
@@ -357,124 +331,144 @@ def open_entries(self, cr, uid, ids, context=None):
-        'remaining_value': fields.float('Next Period Depreciation', digits_compute=dp.get_precision('Account'),required=True),
-        'depreciated_value': fields.float('Amount Already Depreciated', required=True),
-        'depreciation_date': fields.date('Depreciation Date', select=1),
-        'move_id': fields.many2one('account.move', 'Depreciation Entry'),
-        'move_check': fields.function(_get_move_check, method=True, type='boolean', string='Posted', store=True)
-    }
-
-    def create_move(self, cr, uid, ids, context=None):
-        context = dict(context or {})
-        can_close = False
-        asset_obj = self.pool.get('account.asset.asset')
-        period_obj = self.pool.get('account.period')
-        move_obj = self.pool.get('account.move')
-        move_line_obj = self.pool.get('account.move.line')
-        currency_obj = self.pool.get('res.currency')
+    name = fields.Char(string='Depreciation Name', required=True, index=True)
+    sequence = fields.Integer(required=True)
+    asset_id = fields.Many2one('account.asset.asset', string='Asset', required=True, ondelete='cascade')
+    parent_state = fields.Selection(related='asset_id.state', string='State of Asset')
+    amount = fields.Float(string='Current Depreciation', digits=dp.get_precision('Account'), required=True)
+    remaining_value = fields.Float(string='Next Period Depreciation', digits=dp.get_precision('Account'), required=True)
+    depreciated_value = fields.Float(string='Amount Already Depreciated', required=True)
+    depreciation_date = fields.Date('Depreciation Date', index=True)
+    move_id = fields.Many2one('account.move', string='Depreciation Entry')
+    move_check = fields.Boolean(compute='_get_move_check', string='Posted', track_visibility='always', store=True)
+
+    @api.one
+    @api.depends('move_id')
+    def _get_move_check(self):
+        self.move_check = bool(self.move_id)
+
+    @api.multi
+    def create_move(self):
         created_move_ids = []
         asset_ids = []
-        for line in self.browse(cr, uid, ids, context=context):
-            depreciation_date = context.get('depreciation_date') or line.depreciation_date or time.strftime('%Y-%m-%d')
-            period_ids = period_obj.find(cr, uid, depreciation_date, context=context)
-            company_currency = line.asset_id.company_id.currency_id.id
-            current_currency = line.asset_id.currency_id.id
-            context.update({'date': depreciation_date})
-            amount = currency_obj.compute(cr, uid, current_currency, company_currency, line.amount, context=context)
-            sign = (line.asset_id.category_id.journal_id.type == 'purchase' and 1) or -1
+        for line in self:
+            depreciation_date = self._context.get('depreciation_date') or line.depreciation_date or fields.Date.context_today(self)
+            periods = self.env['account.period'].find(depreciation_date)
+            self = self.with_context(date=depreciation_date)
```

Comment:
The context of a recordset(s) is captured in its environment, and by design the context associated to an environment cannot be modified, otherwise it would introduce inconsistencies in the cache.